src/server.rs

```diff
@@ -63,3 +65,12 @@ async fn register_device(mut req: tide::Request<State>) -> tide::Result<tide::Re
 
     Ok(tide::Response::new(tide::StatusCode::Ok))
 }
+
+/// Notifies a single device with a visible notification.
+async fn notify_device(mut req: tide::Request<State>) -> tide::Result<tide::Response> {
+    let token_encrypted = req.body_string().await?; // or body_bytes for Vec<u8>
+    info!("Got direct notification for {token_encrypted}.");
+    // TODO decrypt token with pgp and notify that it got new messages.
```

Comment:
for a first landing of PRs we could also not do the encryption/decryption ... the chatmail instance then sees the cleartext token but it's unclear anyway how much this is a problem.  I regard the core2notification-server encryption as an optimization that could come after it works (maybe even on a new endpoint then).